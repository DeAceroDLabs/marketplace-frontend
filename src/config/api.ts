import { API, Category, Product, ZipCodeLocation } from "./api.types";
import axios from "axios";
import { USER_API } from "./user.api.types";

const fetchData = (url: string) => {
  async function fetchUrl() {
    const response = await fetch(url);
    const json = await response.json();
    return json;
  }
  return fetchUrl();
};

const postRequest = async (url: string, body: any) => {
  const response = await axios.post(USER_API.createUser, body);
  return response;
};

const fetchCategories = (): Promise<Category[]> => {
  return fetchData(API.categories);
};

const fetchProducts = (categoryId: number): Promise<Product[]> => {
  return fetchData(API.products(categoryId));
};

const getLocation = (zipCode: string): Promise<ZipCodeLocation> => {
  return fetchData(API.location(zipCode)).then((data) => {
    if (data["location"]) {
      const neighborhoodsObject = data["location"]["neighborhood"];
      const neighborhoods = Object.keys(neighborhoodsObject);
      const neighborhoodsArray = neighborhoods.map((neighborhood) => {
        return {
          value: neighborhood,
          label: neighborhoodsObject[neighborhood],
        };
      });
      data["location"]["neighborhood"] = neighborhoodsArray;
    }
    return data;
  });
};

const searchProducts = (query: string): Promise<any> => {
  const body = {
    q: query,
  };
  return postRequest(API.search, body);
};

export {
  postRequest,
  fetchCategories,
  fetchProducts,
  searchProducts,
  getLocation,
};
