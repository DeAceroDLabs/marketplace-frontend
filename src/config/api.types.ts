export const API = {
  categories: "categories",
  products: (categoryId: number) => `categories/${categoryId}/products`,
  search: "/search",
  location: (zipCode: string) => `locations/postalCode/${zipCode}`,
};

export type Category = {
  categoryId: number;
  categoryName: string;
  imageUrl: string;
};

export type Article = {
  articleId: string;
};

export type Product = {
  productId: number;
  productName: string;
  categoryName: string;
  categoryId: number;
  articles: Article[];
  categoryImageUrl: string;
};

export type ZipCodeLocation = {
  countryName: string;
  location: Location;
};

export type Location = {
  cityId: string;
  city: string;
  stateId: string;
  state: string;
  neighborhood: Neighborhood[];
};

export type Neighborhood = {
  value: string;
  label: string;
};
