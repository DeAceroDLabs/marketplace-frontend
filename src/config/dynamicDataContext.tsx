import React from "react";

const defaultForm = {
  dynamicData: {} as any,
  setDynamicData: (data: any) => {},
};

export const DynamicDataContext = React.createContext(defaultForm);

export const DynamicDataProvider = DynamicDataContext.Provider;

export default DynamicDataContext;
