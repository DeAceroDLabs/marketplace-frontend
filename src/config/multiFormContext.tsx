import { Form } from "forms/form.types";
import React from "react";

const defaultForm = {
  forms: [] as Form[],
  activeForm: {} as Form,
  setActiveForm: (form: Form) => {},
};

const MultiFormContext = React.createContext(defaultForm);

export const MultiFormProvider = MultiFormContext.Provider;

export default MultiFormContext;
