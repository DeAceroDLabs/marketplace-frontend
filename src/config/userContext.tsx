import React from "react";

const defaultUser = {
  username: "",
  setUser: (username: string) => {},
};

const UserContext = React.createContext(defaultUser);

export const UserProvider = UserContext.Provider;

export default UserContext;
