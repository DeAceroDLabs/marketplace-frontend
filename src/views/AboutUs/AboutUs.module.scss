@use 'styles/colors';
@use 'styles/typography';

.about-us-container {
    margin-bottom: 15%;
}

.image-content {
    display: flex;
    flex-direction: column;
    justify-content: center;
    height: 100%;
    text-align: left;
}

.image-content::after {
    content: '';
    background: rgba(45, 88, 35, 0.7);
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background: linear-gradient(180deg, rgba(0, 0, 0, 0.65) 19.38%, rgba(0, 0, 0, 0.3) 52.15%, rgba(0, 0, 0, 0.05) 88.61%);
}

.image-text {
    position: relative;
    font-size: typography.$body-m;
    z-index: 1;
    color: colors.$color-background-01;
    margin: auto auto;
    width: 80%;
    
}

.about-us-content{
    font-size: typography.$body-m;
    text-align: left;
}
