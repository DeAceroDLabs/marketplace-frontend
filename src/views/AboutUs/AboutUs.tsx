import React from "react";
import { useEffect } from "react";
import styles from "./AboutUs.module.scss";
import BgSection from "components/common/BgSection";
import Section from "components/common/Section";

const AboutUs: React.FunctionComponent = () => {
  useEffect(() => {
    window.scrollTo(0, 0);
  }, []);

  const imageContainer = (
    <BgSection bgImage="/background.png" position="middle">
      <div className={styles["image-content"]}>
        <div className={styles["image-text"]}>
          <div>
            Li Europan lingues es membres del sam familie. Lor separat
            existentie es un myth. Por scientie, musica, sport etc., li tot
            Europa usa li sam vocabularium. Li lingues differe solmen in li
            grammatica, li pronunciation e li plu commun vocabules. Omnicos
            directe al desirabilit? de un nov lingua franca: on refusa continuar
            payar custosi traductores. It solmen va esser necessi far uniform
            grammatica, pronunciation e plu sommun paroles.
          </div>
          <br />
          <div>
            Li Europan lingues es membres del sam familie. Lor separat
            existentie es un myth. Por scientie, musica, sport etc., li tot
            Europa usa li sam vocabularium. Li lingues differe solmen in.
          </div>
        </div>
      </div>
    </BgSection>
  );

  return (
    <div className={styles["about-us-container"]}>
      <Section title="Conócenos" titleSize="large" widthSize="medium"></Section>
      <div>{imageContainer}</div>
      <Section title="Quiénes somos" titleSize="large" widthSize="medium">
        <div className={styles["about-us-content"]}>
          <div>
            Li Europan lingues es membres del sam familie. Lor separat
            existentie es un myth. Por scientie, musica, sport etc., li tot
            Europa usa li sam vocabularium. Li lingues differe solmen in li
            grammatica, li pronunciation e li plu commun vocabules. Omnicos
            directe al desirabilit? de un nov lingua franca: on refusa continuar
            payar custosi traductores. It solmen va esser necessi far uniform
            grammatica, pronunciation e plu sommun paroles.
          </div>
          <br />
          <div>
            Li Europan lingues es membres del sam familie. Lor separat
            existentie es un myth. Por scientie, musica, sport etc., li tot
            Europa usa li sam vocabularium. Li lingues differe solmen in.
          </div>
        </div>
      </Section>
    </div>
  );
};

export default AboutUs;
