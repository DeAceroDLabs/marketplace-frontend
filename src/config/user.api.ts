import { postRequest } from "./api";
import { CreateUser, USER_API } from "./user.api.types";

const createUser = (user: CreateUser) => {
  return postRequest(USER_API.createUser, user);
};
export { createUser };
