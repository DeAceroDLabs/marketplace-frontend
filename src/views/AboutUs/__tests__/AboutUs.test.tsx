import renderer from "react-test-renderer";
import AboutUs from "../AboutUs";
import { render, waitFor } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { UserProvider } from "config/userContext";

window.scrollTo = jest.fn();

describe("Frequent asked questions", () => {
  const mockUSer = {
    username: "mock_user",
    setUser: jest.fn(),
  };
  const renderView = () =>
    render(
      <UserProvider value={mockUSer}>
        <MemoryRouter>
          <AboutUs />
        </MemoryRouter>
      </UserProvider>
    );

  it("renders AboutUs with no issue", () => {
    const component = renderer.create(<AboutUs />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders AboutUs scroll up", async () => {
    renderView();
    await waitFor(() => {
      expect(window.scrollTo).toBeCalled();
    });
  });
});
