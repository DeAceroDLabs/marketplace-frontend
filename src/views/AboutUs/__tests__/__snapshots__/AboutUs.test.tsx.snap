// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Frequent asked questions renders AboutUs with no issue 1`] = `
<div
  className="about-us-container"
>
  <div
    className="container"
  >
    <div
      className="title margin-title"
    >
      Conócenos
       
    </div>
    <div
      className="body"
    >
       
       
    </div>
  </div>
  <div>
    <div
      className="horizontal"
    >
      <img
        alt=""
        className="bgImage"
        src="/background.png"
      />
      <div
        className="content transparent middle"
      >
        <div
          className="image-content"
        >
          <div
            className="image-text"
          >
            <div>
              Li Europan lingues es membres del sam familie. Lor separat existentie es un myth. Por scientie, musica, sport etc., li tot Europa usa li sam vocabularium. Li lingues differe solmen in li grammatica, li pronunciation e li plu commun vocabules. Omnicos directe al desirabilit? de un nov lingua franca: on refusa continuar payar custosi traductores. It solmen va esser necessi far uniform grammatica, pronunciation e plu sommun paroles.
            </div>
            <br />
            <div>
              Li Europan lingues es membres del sam familie. Lor separat existentie es un myth. Por scientie, musica, sport etc., li tot Europa usa li sam vocabularium. Li lingues differe solmen in.
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="title margin-title"
    >
      Quiénes somos
       
    </div>
    <div
      className="body"
    >
       
      <div
        className="about-us-content"
      >
        <div>
          Li Europan lingues es membres del sam familie. Lor separat existentie es un myth. Por scientie, musica, sport etc., li tot Europa usa li sam vocabularium. Li lingues differe solmen in li grammatica, li pronunciation e li plu commun vocabules. Omnicos directe al desirabilit? de un nov lingua franca: on refusa continuar payar custosi traductores. It solmen va esser necessi far uniform grammatica, pronunciation e plu sommun paroles.
        </div>
        <br />
        <div>
          Li Europan lingues es membres del sam familie. Lor separat existentie es un myth. Por scientie, musica, sport etc., li tot Europa usa li sam vocabularium. Li lingues differe solmen in.
        </div>
      </div>
       
    </div>
  </div>
</div>
`;
